src/org/armedbear/lisp/gray-streams.lisp

```diff
@@ -215,6 +215,15 @@
 (defgeneric gray-pathname (pathspec))
 (defgeneric gray-truename (filespec))
 
+(defun assert-stream (stream)
+  (if (gray-streamp stream)
+      t
+      (error 'type-error :datum stream :expected-type 'stream)))
+
+(defun bug-or-error (stream fun)
```

Comment:
TODO check if there is an ANSI error we can throw here…

Reply:
Are you referring to line 221 or to the bug-or-error function?

Reply:
I think most the "Exceptional situations" require type-errors. I'll scan through them and summarize.

Reply:
Requires type-error if not stream: [stream-element-type](https://novaspec.org/cl/f_stream-element-type), [input-stream-p](https://novaspec.org/cl/f_input-stream-p), [output-stream-p](https://novaspec.org/cl/f_output-stream-p), [interactive-stream-p](https://novaspec.org/cl/f_interactive-stream-p), or [open-stream-p](https://novaspec.org/cl/f_open-stream-p)

Doesn't specify "Exceptional situations," but the argument is listed as a stream: (close](https://novaspec.org/cl/f_close)

Doesn't specify "Exceptional situations," but the argument is listed as a pathspec, which is generally a string, a pathname or a stream: (pathname](https://novaspec.org/cl/f_pathname) and (truename](https://novaspec.org/cl/f_truename)

Reply:
Sorry, the conversation thread above got a little confusing. I think I replied while you where editing your original comment.

To summarize my response, I think the type errors are more correct then STREAM-ERROR because the "Exceptional Situation" sections in the above.

Reply:
I agree with your assessment.  I will merge this when I get a chance.

Reply:
Thank you!